### PR TITLE
Table of content for the docs

### LINKED ISSUE #1074: table of content for the docs
We have more and more documentation it would be nice to have a table of content in the docs for easier navigation.
For example user guide is quite long and there are many sections.

### PR DESCRIPTION
**Pull request by csordasmarton**
Closes #1074

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -3,7 +3,27 @@
...
+  * [Analysis Failures](#analysis-failures)
+* [Step 3: Store analysis results in a CodeChecker DB and visualize results](#step-3)
+* [Step 4: Fine tune Analysis configuration](#step-4)
+  * [Ignore modules from your analysis](#ignore-modules)
+  * [Enable/Disable Checkers](#enable-disable-checkers)
+  * [Identify files that failed analysis](#identify-files)
+* [Step 5: Integrate CodeChecker into your CI loop](#step-5)
+  * [Storing daily runs](#storing-daily-runs)
+  * [Store each analysis in a new run](#storing-new-runs)
+    * [Example](#storing-new-runs-example)
```
Comment (gyorb) [inline docs/usage.md]: Do we want to include `Example` into the table of content? I've seen it at multiple places but I'm not sure we should include them.

File: docs/user_guide.md [hunk 2]
```diff
@@ -1,4 +1,57 @@
...
+    * [Using SQLite for database](#sqlite)
+  * [`checkers`](#checkers)
+  * [`analyzers`](#analyzers)
+  * [`server`](#server)
+    * [Creating a public server](#public-server)
+    * [Configuring database and server settings' location](#server-settings)
+    * [Master superuser and authentication forcing](#auth-force)
+    * [Enfore secure socket (SSL)](#ssl)
+    * [Managing running servers](#managing-running-servers)
+  * [cmd](#cmd)
```
Comment (gyorb) [inline docs/user_guide.md]: This is not highlighted as the others.

File: docs/user_guide.md [hunk 3]
```diff
@@ -1,4 +1,57 @@
...
+    * [List runs (`runs`)](#cmd-runs)
```
Comment (gyorb) [inline docs/user_guide.md]: I think we should change the order here for the subcommands.

`runs` (List of runs) 
`results` (Analysis results)
...

File: docs/usage.md (hunk 1 above)
Comment (whisperity) [reply docs/usage.md]: It's a direct link to a code snippet only... well it's debateable. Including it makes the TOS easier, and perhaps later on, automatic to maintain... but I don't see a reason in including this specifically either.

File: docs/user_guide.md (hunk 3 above)
Comment (whisperity) [reply docs/user_guide.md]: The order is taken from how the help is generated by `argparse` which in turn, I think, takes the subcommands in the order they are defined in the source file.

Comment (gyorb) [reply docs/user_guide.md]: I meant not the order of the items (`runs`, `results`) but the order of the command name and the short description.

Comment (whisperity) [reply docs/user_guide.md]: Got it. That's a good idea. 😉

File: docs/usage.md (hunk 1 above)
Comment (csordasmarton) [reply docs/usage.md]: If the user knows about the feature but don't know how to use it, he checks the TOC and sees that there is an example for this feature and he can jump to it. I think we should leave it in the TOC.

**Comment by whisperity** [comment]
@Xazax-hun There doesn't seem to be. There are few things all around GitHub that seemingly do this functionality, for example NPM packages and such at render time, but none seems to really work in a way that they generate the TOC into the markdown. And this also increases the usability of the docs if viewed on GitHub too.

**Pull request merged**